firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -190,21 +191,24 @@ public void finalizeReports(String currentSessionId) {
     return allReports;
   }
 
-  private static boolean isHighPriorityEvent(String eventFileName) {
-    return eventFileName.endsWith(PRIORITY_EVENT_SUFFIX);
+  private static boolean isHighPriorityEventFile(String fileName) {
+    return fileName.startsWith(EVENT_FILE_NAME_PREFIX) && fileName.endsWith(PRIORITY_EVENT_SUFFIX);
   }
 
-  private String generateEventFilename(boolean isHighPriority) {
-    final String eventNumber =
-        String.format(Locale.US, EVENT_COUNTER_FORMAT, eventCounter.getAndIncrement());
-    final String prioritySuffix = isHighPriority ? PRIORITY_EVENT_SUFFIX : NORMAL_EVENT_SUFFIX;
-    return EVENT_FILE_NAME_PREFIX + eventNumber + prioritySuffix;
+  private static boolean isNormalPriorityEventFile(File dir, String name) {
+    return name.startsWith(EVENT_FILE_NAME_PREFIX) && !name.endsWith(PRIORITY_EVENT_SUFFIX);
```

Comment:
I'm prefer startsWith and endsWith here, though I could also see some folks preferring a regex match

Reply:
Pretty certain `startsWith` and `endsWith` are more efficient than a regex would be, which is why I used them.